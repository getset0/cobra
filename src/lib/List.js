import styled, { css } from "styled-components";

const List = styled.ul`
  list-style: none;
  margin: 0;
  padding: 0px;

  > li {
    display: flex;
    justify-content: flex-start;
    ${props => props.center && css`
      justify-content: center;
    `}
    ${props => props.spaceAround && css`
      justify-content: space-around;
    `}
    ${props => props.spaceBetween && css`
      justify-content: space-between;
    `}
    ${props => props.end && css`
      justify-content: flex-end;
    `}
  }
`;

export default List;
