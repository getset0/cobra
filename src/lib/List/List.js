import styled, { css } from "styled-components";
import PropTypes from "prop-types";

const List = styled.ul`
  list-style: none;
  margin: 0;
  padding: 0px;
  ${props =>
    props.orientation === "horizontal" &&
    css`
      display: flex;
    `}

  > li {
    display: flex;
    justify-content: flex-start;
    ${props =>
    props.orientation === "horizontal" &&
      css`
        border: none;
      `}
    ${props =>
    props.center &&
      css`
        justify-content: center;
      `}
    ${props =>
    props.spaceAround &&
      css`
        justify-content: space-around;
      `}
    ${props =>
    props.spaceBetween &&
      css`
        justify-content: space-between;
      `}
    ${props =>
    props.end &&
      css`
        justify-content: flex-end;
      `}
  }
`;

List.propTypes = {
  center: PropTypes.bool,
  spaceAround: PropTypes.bool,
  spaceBetween: PropTypes.bool,
  end: PropTypes.bool,
  orientation: PropTypes.string
};

List.defaultProps = {
  orientation: "vertical"
};

export default List;
