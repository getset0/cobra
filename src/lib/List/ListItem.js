import styled, { css } from "styled-components";
import PropTypes from "prop-types";
import { theme } from "../constants";

const ListItem = styled.li`
  padding: 12px 10px 12px 10px;
  ${props => props.onClick && css`
    cursor: pointer;
  `}
  border-top: 1px solid ${props => props.theme.lightGrayColor};
  &:first-child {
    border-top: 0;
  }
`;

ListItem.propTypes = {
  onClick: PropTypes.func,
  theme: PropTypes.object
};

ListItem.defaultProps = {
  theme: theme
};

export default ListItem;
