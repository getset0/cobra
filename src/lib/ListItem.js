import styled, { css } from "styled-components";

const ListItem = styled.li`
  padding: 12px 10px 12px 10px;
  ${props => props.onClick && css`
    cursor: pointer;
  `}
  border-top: 1px solid ${props => props.theme.lightGrayColor};
  &:first-child {
    border-top: 0;
  }
`;

export default ListItem;
