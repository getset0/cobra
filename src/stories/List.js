import React from "react";
import { storiesOf } from "@storybook/react";
import { withInfo } from "@storybook/addon-info";
import { withBackgrounds } from "@storybook/addon-backgrounds";
import { backgroundOptions, testImageSrc } from "./util";
import { Image, List, ListItem } from "../lib";

const CustomListItem = () => (
  <React.Fragment>
    <Image src={testImageSrc} width={200} height={100} />
    <div
      style={{
        display: "flex",
        alignItems: "center",
        marginLeft: "10px"
      }}
    >
        Text Text Text Text Text Text Text Text Text Text Text Text Text Text Text
        Text Text Text
    </div>
  </React.Fragment>
);

storiesOf("List", module)
  .addDecorator(withInfo)
  .addDecorator(withBackgrounds(backgroundOptions))
  .add("Text", () => (
    <List>
      <ListItem>Item</ListItem>
      <ListItem>Item</ListItem>
      <ListItem>Item</ListItem>
    </List>
  ))
  .add("Horizontal", () => (
    <List orientation="horizontal">
      <ListItem>Item</ListItem>
      <ListItem>Item</ListItem>
      <ListItem>Item</ListItem>
    </List>
  ))
  .add("Centered", () => (
    <List center>
      <ListItem>Item</ListItem>
      <ListItem>Item</ListItem>
      <ListItem>Item</ListItem>
    </List>
  ))
  .add("Custom Component", () => (
    <List>
      <ListItem>
        <CustomListItem />
      </ListItem>
      <ListItem>
        <CustomListItem />
      </ListItem>
      <ListItem>
        <CustomListItem />
      </ListItem>
    </List>
  ))
  .add("Custom Component Space Around", () => (
    <List spaceAround>
      <ListItem>
        <CustomListItem />
      </ListItem>
      <ListItem>
        <CustomListItem />
      </ListItem>
      <ListItem>
        <CustomListItem />
      </ListItem>
    </List>
  ))
  .add("Custom Component Space Between", () => (
    <List spaceBetween>
      <ListItem>
        <CustomListItem />
      </ListItem>
      <ListItem>
        <CustomListItem />
      </ListItem>
      <ListItem>
        <CustomListItem />
      </ListItem>
    </List>
  ));


