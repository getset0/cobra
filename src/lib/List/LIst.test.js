import React from "react";
import ListItem from "./ListItem";
import List from "./List";
import "jest-styled-components";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

const ListOfItems = () =>
  <List>
    <ListItem>1</ListItem>
    <ListItem>2</ListItem>
    <ListItem>3</ListItem>
  </List>;

it("renders List with ListItems without crashing", () => {
  mount(ListOfItems());
});

it("renders List with ListItems correctly", () => {
  const wrapper = shallow(ListOfItems());
  const tree = toJson(wrapper);
  expect(tree).toMatchSnapshot();
});
