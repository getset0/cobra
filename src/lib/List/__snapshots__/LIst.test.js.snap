// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders List with ListItems correctly 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": false,
        "lastClassName": "daEYVW",
        "rules": Array [
          "
  list-style: none;
  margin: 0;
  padding: 0px;

  > li {
    display: flex;
    justify-content: flex-start;
    ",
          [Function],
          "
    ",
          [Function],
          "
    ",
          [Function],
          "
    ",
          [Function],
          "
  }
",
        ],
      },
      "displayName": "styled.ul",
      "propTypes": Object {
        "center": [Function],
        "end": [Function],
        "spaceAround": [Function],
        "spaceBetween": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "ul",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  <styled.li
    theme={
      Object {
        "blackColor": "#000000",
        "darkBlueColor": "#02469B",
        "darkGrayColor": "#4A4A4A",
        "darkestBlueColor": "#02254D",
        "darkestGrayColor": "#303030",
        "greenColor": "#76B041",
        "lightBlueColor": "#0190D0",
        "lightGrayColor": "#D8D8D8",
        "lightestBlueColor": "#00FEFE",
        "mediumGrayColor": "#979797",
        "primaryColor": "#1676B7",
        "secondaryColor": "#17BEBB",
        "whiteBackgroundColor": "#fdfdfd",
        "whiteColor": "#ffffff",
        "yellowColor": "#FFC914",
      }
    }
  >
    1
  </styled.li>
  <styled.li
    theme={
      Object {
        "blackColor": "#000000",
        "darkBlueColor": "#02469B",
        "darkGrayColor": "#4A4A4A",
        "darkestBlueColor": "#02254D",
        "darkestGrayColor": "#303030",
        "greenColor": "#76B041",
        "lightBlueColor": "#0190D0",
        "lightGrayColor": "#D8D8D8",
        "lightestBlueColor": "#00FEFE",
        "mediumGrayColor": "#979797",
        "primaryColor": "#1676B7",
        "secondaryColor": "#17BEBB",
        "whiteBackgroundColor": "#fdfdfd",
        "whiteColor": "#ffffff",
        "yellowColor": "#FFC914",
      }
    }
  >
    2
  </styled.li>
  <styled.li
    theme={
      Object {
        "blackColor": "#000000",
        "darkBlueColor": "#02469B",
        "darkGrayColor": "#4A4A4A",
        "darkestBlueColor": "#02254D",
        "darkestGrayColor": "#303030",
        "greenColor": "#76B041",
        "lightBlueColor": "#0190D0",
        "lightGrayColor": "#D8D8D8",
        "lightestBlueColor": "#00FEFE",
        "mediumGrayColor": "#979797",
        "primaryColor": "#1676B7",
        "secondaryColor": "#17BEBB",
        "whiteBackgroundColor": "#fdfdfd",
        "whiteColor": "#ffffff",
        "yellowColor": "#FFC914",
      }
    }
  >
    3
  </styled.li>
</StyledComponent>
`;
